src/atoms/sdp_cone/geom_mean_epicone.jl

```diff
@@ -0,0 +1,120 @@
+#############################################################################
+# geom_mean_epicone.jl
+# Constrains T to
+#   A #_t B ⪯ T
+# where:
+#   * A #_t B is the t-weighted geometric mean of A and B:
+#     A^{1/2} (A^{-1/2} B A^{-1/2})^t A^{1/2}
+#     Parameter t should be in [-1, 0] or [1, 2].
+#   * Constraints A ⪰ 0, B ⪰ 0 are added.
+# All expressions and atoms are subtypes of AbstractExpr.
+# Please read expressions.jl first.
+#
+# Note on fullhyp: GeomMeanEpiCone will always return a full epigraph cone
+# (unlike GeomMeanHypoCone) and so this parameter is not really used.  It is
+# here just for consistency with the GeomMeanHypoCone function.
+#
+#REFERENCE
+#   Ported from CVXQUAD which is based on the paper: "Lieb's concavity
+#   theorem, matrix geometric means and semidefinite optimization" by Hamza
+#   Fawzi and James Saunderson (arXiv:1512.03401)
+#############################################################################
+
+struct GeomMeanEpiCone
+    A::AbstractExpr
+    B::AbstractExpr
+    t::Rational
+    size::Tuple{Int, Int}
+
+    function GeomMeanEpiCone(A::AbstractExpr, B::AbstractExpr, t::Rational, fullhyp::Bool=true)
+        if size(A) != size(B)
+            error("A and B must be the same size")
+        end
+        n = size(A)[1]
+        if size(A) != (n, n)
+            error("A and B must be square")
+        end
+        if t < -1 || (t > 0 && t < 1) || t > 2
+            error("t must be in the range [-1, 0] or [1, 2]")
+        end
+        return new(A, B, t, (n, n))
+    end
+
+    GeomMeanEpiCone(A::Value,        B::AbstractExpr, t::Rational, fullhyp::Bool=true) = GeomMeanEpiCone(Constant(A), B, t, fullhyp)
+    GeomMeanEpiCone(A::AbstractExpr, B::Value,        t::Rational, fullhyp::Bool=true) = GeomMeanEpiCone(A, Constant(B), t, fullhyp)
+    GeomMeanEpiCone(A::Value,        B::Value,        t::Rational, fullhyp::Bool=true) = GeomMeanEpiCone(Constant(A), Constant(B), t, fullhyp)
+end
+
+struct GeomMeanEpiConeConstraint <: Constraint
+    head::Symbol
+    id_hash::UInt64
+    T::AbstractExpr
+    cone::GeomMeanEpiCone
+
+    function GeomMeanEpiConeConstraint(T::AbstractExpr, cone::GeomMeanEpiCone)
+        if size(T) != cone.size
+            error("T must be size $(cone.size)")
+        end
+        id_hash = hash((cone.A, cone.B, cone.t, :GeomMeanEpiCone))
+        return new(:GeomMeanEpiCone, id_hash, T, cone)
+    end
+
+    GeomMeanEpiConeConstraint(T::Value, cone::GeomMeanEpiCone) = GeomMeanEpiConeConstraint(Constant(T), cone)
+end
+
+in(T, cone::GeomMeanEpiCone) = GeomMeanEpiConeConstraint(T, cone)
+
+function AbstractTrees.children(constraint::GeomMeanEpiConeConstraint)
+    return (constraint.T, constraint.cone.A, constraint.cone.B, "t=$(constraint.cone.t)")
+end
+
+# FIXME what is the meaning of the vexity of a variable? of a constraint?  Is this correct?
```

Comment:
sorry for the delay. So the meaning of vexity of a variable is just:

* ConstVexity iff it's `fix!`'d and should be treated as a constant
* AffineVexity otherwise

That is, we treat a variable `x` like the function `f(x) = x` for the purposes of deciding the vexity (and like `f(x) = c` if it's a constant).

The vexity of a `constraint` is just to determine is the constraint of the form `f(x) <= 0` for convex `f` (where `<=` might be with respect to a general cone, not necessarily just in R or C).

Reply:
so to answer the question of "is this correct", I'm not 100% sure... the question is if `A` and `B` were functions of some underlying vector of variables `x` what conditions are needed on `A` and `B` such that
```
A^{1/2} (A^{-1/2} B A^{-1/2})^t A^{1/2}
```
is a convex function of `x`?

i.e. a possible "condition" would be `A` and `B` must be convex functions (just as an example, I don't know what the conditions are here).